bin/dfx-state-install

```diff
@@ -17,7 +17,7 @@ help_text() {
 # Source the clap.bash file ---------------------------------------------------
 source "$SOURCE_DIR/clap.bash"
 # Define options
-clap.define short=f long=state desc="The state file to save to" variable=DFX_STATE_FILE default="state.zip"
+clap.define short=f long=state desc="The state file to save to" variable=DFX_STATE_FILE default="$(dfx cache show)/state.zip"
```

Comment:
We should probably check if the file exists and give a useful error if it doesn't.

Reply:
Yes.  Actually, I think it would be best to have a search path.  Save locally, restore locally if there is a local snapshot, else get the snapshot from a global cache, on the basis that the snapshots are going to be the same for most branches.  No point in having lots of local copies, or making the user type a global path every time they want to spin up an environment.  That sort of API is called FRICTION. :-)

Reply:
This should do the trick.  I don't have time to finish the PR now but when I do I'll ask for a review.  https://github.com/dfinity/snsdemo/pull/115/files